### ISSUE TITLE
twisted.conch.client.knownhosts uses Crypto.Hash.HMAC, could easily use stdlib hmac instead

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3849]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#3849|
|Type|enhancement|
|Created|2009-05-20 17:51:34Z|
|Branch|https://github.com/twisted/twisted/tree/conch-hmac-3849|

The former of these two is actually a fork of the latter.  Just switching knownhosts to the stdlib version won't make any part of Conch work without PyCrypto, but I still don't see a particularly good reason to use the PyCrypto version of this library rather than the stdlib version.

Attachments:

* [stdlib_hmac.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/80c/80c48ed61ca64514690e96a5659cddd90267e8f3/4718652158a7f192fc02fdab2fd907c9d05d5300.patch) (703 bytes) - added by jtanis on 2010-02-21 22:31:28Z - Replaced Crypto.Hash.HMAC with stdlib hmac as stated

<details><summary>Searchable metadata</summary>

```
trac-id__3849 3849
type__enhancement enhancement
reporter__exarkun exarkun
priority__low low
milestone__ 
branch__branches_conch_hmac_3849 branches/conch-hmac-3849
branch_author__therve therve
status__closed closed
resolution__fixed fixed
component__conch conch
keywords__easy easy
time__1242841894000000 1242841894000000
changetime__1266867363000000 1266867363000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|<img alt="jtanis's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|jtanis set owner to jtanis<br>jtanis set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="jtanis's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|jtanis removed owner<br>jtanis set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve commented|
|-|-|

(In [28343]) Branching to 'conch-hmac-3849'

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)|@therve set owner to @therve|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)|@therve removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="terrycojones's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_6"></a>|terrycojones commented|
|-|-|

Looks good. trial twisted passes.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_7"></a>|@therve set status to `closed`|
|-|-|

(In [28407]) Merge conch-hmac-3849

Authors: jtanis, therve
Reviewer: terrycojones
Fixes [#3849](https://github.com/twisted/twisted/issues/3849)

Remove usage of Crypto.Hash from twisted.conch.client.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
